plugins/kotlin/j2k/shared/src/org/jetbrains/kotlin/nj2k/conversions/StreamConversion.kt

```diff
@@ -0,0 +1,38 @@
+// Copyright 2000-2024 JetBrains s.r.o. and contributors. Use of this source code is governed by the Apache 2.0 license.
+package org.jetbrains.kotlin.nj2k.conversions
+
+import org.jetbrains.kotlin.analysis.api.KtAnalysisSession
+import org.jetbrains.kotlin.nj2k.NewJ2kConverterContext
+import org.jetbrains.kotlin.nj2k.RecursiveConversion
+import org.jetbrains.kotlin.nj2k.callOn
+import org.jetbrains.kotlin.nj2k.tree.*
+
+class StreamConversion(context: NewJ2kConverterContext) : RecursiveConversion(context) {
+
+    context(KtAnalysisSession)
+    override fun applyToElement(element: JKTreeElement): JKTreeElement {
+        if (element !is JKQualifiedExpression) return recurse(element)
+        if (element.selector !is JKCallExpressionImpl) return recurse(element)
+
+        val target = (element.selector as JKCallExpressionImpl).identifier.fqName
+
+        // TODO: Handle other cases too.
+        return recurse(
+            when (target) {
+                "java.lang.String.lines" -> convertStringLines(element)
+                else -> element
+            }.withFormattingFrom(element)
+        )
+    }
+
+    private fun convertStringLines(element: JKQualifiedExpression): JKTreeElement {
+        val lineSequence = "kotlin.text.lineSequence"
+        val asStream = "kotlin.streams.asStream"
+
+        context.importStorage.addImport(asStream)
+
+        return element.receiver.copyTreeAndDetach()
+            .callOn(symbolProvider.provideMethodSymbol(lineSequence))
+            .callOn(symbolProvider.provideMethodSymbol(asStream))
```

Comment:
It's better to add a test with some comments, they should be preserved. `withFormattingFrom` call may be necessary.

Reply:
I was trying to keep other cases in mind too, so I called `withFormattingFrom` on a higher level. Should I move it down here?

```kotlin
return recurse(
            when (target) {
                "java.lang.String.lines" -> convertStringLines(element)
                else -> element
            }.withFormattingFrom(element)
        )
```

Reply:
I added a test for formatting.

Reply:
Yep, I have missed the call on a higher level. All good.